src/h5web/providers/hsds/hsds-api.ts

```diff
@@ -182,33 +187,42 @@ export class HsdsApi extends ProviderApi {
     id: HsdsId,
     path: string,
     name: string,
-    depth: number
+    isChild = false
   ): Promise<HsdsEntity<Group>> {
     const { attributeCount, linkCount } = await this.fetchGroup(id);
 
     // Fetch attributes and links in parallel
-    // If recursion depth has been reached, don't fetch links at all
     const [attributes, links] = await Promise.all([
       attributeCount > 0
         ? this.fetchAttributes('groups', id)
         : Promise.resolve([]),
-      linkCount > 0 && depth > 0 ? this.fetchLinks(id) : Promise.resolve([]),
+      !isChild
+        ? linkCount > 0
+          ? this.fetchLinks(id)
+          : Promise.resolve([]) // no links to fetch
+        : Promise.resolve(undefined), // don't fetch nested groups' children
     ]);
 
-    const children = await Promise.all(
-      links.map((link) =>
-        this.resolveLink(link, buildEntityPath(path, link.title), depth - 1)
-      )
-    );
-
-    return {
+    const group: HsdsEntity<Group> = {
       id,
       path,
       name,
       kind: EntityKind.Group,
       attributes: convertHsdsAttributes(attributes),
-      children,
     };
+
+    if (!links) {
```

Comment:
...and put `isChild` here.

The fact that `isChild` leads to undefined `links` that trigger this condition is an unneeded misdirection IMO

Reply:
Ah nice, I get you 👍